include/triton/Dialect/Triton/IR/TritonOps.td

```diff
@@ -460,17 +460,12 @@ def TT_ReshapeOp : TT_Op<"reshape", [Pure,
         If efficient_layout is set, this is a hint that the destination layout should be kept for performance reason.
         The compiler is still free to change it for better performance.
     }];
-    let arguments = (ins TT_Tensor:$src, BoolAttr:$allow_reorder, OptionalAttr<UnitAttr>:$efficient_layout);
+    let arguments = (ins TT_Tensor:$src, UnitAttr:$allow_reorder, UnitAttr:$efficient_layout);
```

Comment:
I liked using `BoolAttr` better as it makes it explicit that this this needs to be set.
I'm always a bit confused with the whether UnitAttr should be Optional or not as it can still be null. I see a mix of both upstream. It's probably fine to remove the optional although I'm not sure it is clearer

Reply:
I also feel `UnitAttr` is inconsistently used... According to [the documentation](https://mlir.llvm.org/docs/DefiningDialects/Operations/#unit-attributes), tho:

> In MLIR, the [unit Attribute](https://mlir.llvm.org/docs/Dialects/Builtin/#unitattr) is special in that it only has one possible value, i.e. it derives meaning from its existence.

[Also](https://mlir.llvm.org/docs/Dialects/Builtin/#unitattr):

> A unit attribute is an attribute that represents a value of unit type. The unit type allows only one value forming a singleton set. This attribute value is used to represent attributes that only have meaning from their existence.

I think making it optional is bad practice, as the functions to check the attribute return `optional<bool>` (instead of the `bool` return type when it is not an optional attribute).

Regarding `BoolAttr` vs `UnitAttr`, the only difference in terms of API is `BoolAttr` must be explicitly specified whereas `UnitAttr` is `None` by default (in addition to the attribute types in the interfaces, ofc). I tend to prefer `UnitAttr`, TBH, but we can keep this `BoolAttr` instead.

If prefered, I can revert `allow_order` type changes and keep `efficient_layout`. ASM format looks better now IMO tho.

Reply:
> ASM format looks better now IMO tho.

fair point, I agree

Reply:
Thanks for the quick review and merge!